python-frontend/src/main/java/org/sonar/python/semantic/SymbolTableBuilder.java

```diff
@@ -517,8 +521,53 @@ private void addSymbolUsage(Name nameTree) {
       SymbolImpl symbol = scope.resolve(nameTree.name());
       // TODO: use Set to improve performances
       if (symbol != null && symbol.usages().stream().noneMatch(usage -> usage.tree().equals(nameTree))) {
-        symbol.addUsage(nameTree, Usage.Kind.OTHER);
+          symbol.addUsage(nameTree, Usage.Kind.OTHER);
       }
     }
+
+    private void resolveParents(ClassDef classDef) {
+      Symbol symbol = classDef.name().symbol();
+      if (symbol == null || !Symbol.Kind.CLASS.equals(symbol.kind())) {
+        return;
+      }
+      ClassSymbolImpl classSymbol = (ClassSymbolImpl) symbol;
+      if (hasMultipleBindingUsages(classSymbol)) {
+        classSymbol.setKind(Symbol.Kind.OTHER);
+        return;
+      }
+      ArgList argList = classDef.args();
+      classSymbol.hasUnresolvedParents = false;
+      if (argList == null) {
+        return;
+      }
+      for (Argument argument : argList.arguments()) {
+        if (!argument.is(Kind.REGULAR_ARGUMENT) || !(((RegularArgument) argument).expression() instanceof HasSymbol)) {
+          classSymbol.hasUnresolvedParents = true;
+          return;
+        }
+        Symbol parentSymbol = ((HasSymbol) ((RegularArgument) argument).expression()).symbol();
+        if (parentSymbol == null) {
+          classSymbol.hasUnresolvedParents = true;
+          return;
+        }
+        if (hasMultipleBindingUsages(parentSymbol)) {
+          classSymbol.hasUnresolvedParents = true;
+          return;
+        }
+        if (parentSymbol.fullyQualifiedName() != null && BuiltinSymbols.all().contains(parentSymbol.fullyQualifiedName())) {
+          classSymbol.addParent(parentSymbol);
+          continue;
+        }
+        if (!Symbol.Kind.CLASS.equals(parentSymbol.kind())) {
```

Comment:
Even here, I think we should always add the found symbol and avoid setting `hasUnresolveParents` to true